### DISCUSSION TITLE
Managing external process's emit

### DISCUSSION THREAD

**Discussion by hyiip** [miguelgrinberg/Flask-SocketIO#1726] [Q&A]
Hi everyone. I am a beginner to SocketIO development. I read from [https://flask-socketio.readthedocs.io/en/latest/deployment.html#emitting-from-an-external-process](url) that I can emit an event from external process. But is there any recommendation on how to manage those event into an main file?
For instance, I have
```python
# event.py
def multiple(a,b):
    socketio = SocketIO(message_queue = "redis://localhost/1")
    output_string = "{a} times {b} is {result}".format(a=a,b=b,result=a*b)
    socketio.emit('multiple', output_string)
...
```

```python
# main.py
...
@socketio.on("connect")
def initialized():
    # ... authorization ...
    socketio.emit("something", data, rooms = "specific_room")

@socketio.on("disconnect")
def disconnect():
    # ... authorization ...
    socketio.emit("something", data, rooms = "specific_room")

@socketio.on('multiple')
def patient_change(data):
    socketio.emit('multiple_client', data, rooms = "specific_room")
...
```
So that when multiple is called, there will be a event emitted. I want that user in specific room can receive the event. I noticed that I am unable to handle event 'multiple' in server side, but only on the client side. Is there any possible way to manage/organize such external event? Or should I not do this?

**Comment by miguelgrinberg** [comment]
I think this may be a misunderstanding on your part on what the "emitting from an external process" feature does. The idea is that you can emit to a client, a room, or all connected clients from the external process. There is no provision to emit from the external process to the server, that is not a feature I intended to support.

If all you want to do in your external process is to emit to a room full of users, then just add `room='your-room-name'` in the emit call. That works in the external process.
